source/agora/common/TransactionPool.d

```diff
@@ -301,6 +301,68 @@ public class TransactionPool
 
         return true;
     }
+
+    /***************************************************************************
+
+        Check if a transaction hash exists in the transaction pool.
+
+        Params:
+            tx = the transaction hash
+
+        Returns:
+            true if the has transaction hash.
+
+    ***************************************************************************/
+
+    public bool hasTransactionHash (const ref Hash tx) @trusted
+    {
+        auto results = this.db.execute("SELECT key, val FROM tx_pool " ~
+            "WHERE key = ?", tx[]);
+
+        if (!results.empty)
+            return true;
+
+        return false;
+    }
+}
+
+/// hasTransactionHash tests
+unittest
+{
+    import agora.consensus.Genesis;
+
+    auto pool = new TransactionPool(":memory:");
+    scope(exit) pool.shutdown();  // note: must call outside destructor
+
+    auto gen_key = getGenesisKeyPair();
+    auto txs = makeChainedTransactions(gen_key, null, 1);
+
+    txs.each!(tx => pool.add(tx));
+    assert(pool.length == txs.length);
+
+    foreach (Transaction tx; txs)
+    {
+        const(Hash) hash = hashFull(tx);
+        assert(pool.hasTransactionHash(hash));
+        pool.remove(tx);
+        assert(!pool.hasTransactionHash(hash));
+    }
+
+    txs.each!(tx => pool.add(tx));
+    assert(pool.length == txs.length);
+    const(Hash) hash = Hash.init;
+    assert(!pool.hasTransactionHash(hash));
+    // 'or 1=1-- SQL Injection Check
```

Comment:
Why is this called an "injection check"? I'm not sure how it's trying to inject anything?

Reply:
I put it in a test.
We receive parameters through API.
Although converted to hash (bitblob!512), we checked for SQL injection attacks.

Reply:
Ah right, I see the comment now. That's a nice idea. I would also (maybe in another pull request) add tests for `pool.add`, making sure injections don't work.

Reply:
Ah, never mind. I just realized `add` takes the transaction, not the hash. Ignore my comment.